### PR TITLE
[WIP] RDF graph functionalities

### PR DESCRIPTION
**Pull request by gallupBenRyan**
Using Stanford Parsing engine, take a JSON file with texts and extract RDF-triple style entities and relationships from the text and compile these into a semantic knowledge graph. These entities/relations are clustered together based on cosine similarities of their (averaged, in the case of multi-word concepts) GloVe embedding vectors. Using OpenKE module, this graph can then be queried using full English language questions and return responses in semantic-triple form. 

Adding reviewers for their visibility, however this update is still a work in progress. More needs to be done in the `rdf_graph.py` file to refine and test the KG induction.

### CODE REVIEW

File: src/rdf_graph/build_graph.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    # output text files that will be used openke for knowledge graph creation
+    # and embedding output .txt files for openke output
+    print('\nSaving final tuples to .txt files for input to OpenKE')
+    knowledge_graph.output_to_openke(timestamp)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Fetcher to retrieve articles '
+                                     'for modeling.')
+    parser.add_argument('-s', '--sources', required=True, choices=['pubmed',
```
Comment (BoltGallup) [inline src/rdf_graph/build_graph.py]: Do we still need this argument, given we're not calling to Pubmed or Arxiv for data?

File: src/rdf_graph/build_graph.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+import argparse
+
+from datetime import datetime
+
+from rdf_graph import RDFGraph
```
Comment (BoltGallup) [inline src/rdf_graph/build_graph.py]: Hey Ben, do we want this code to be able to run from the base of the repo? Or are we running from the 'rdf_graph' folder?

File: src/rdf_graph/build_graph.py (hunk 1 above)
Comment (gallupBenRyan) [reply src/rdf_graph/build_graph.py]: We do not - good call.

And for the record - feel free to take over this branch and make the necessary changes. Not to discourage asking questions - do! - but just to clarify that you don't need my permission to fix things.

File: src/rdf_graph/rdf_graph.py [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+
+    @staticmethod
+    def get_unique_entities(tuples, top_n_ent):
+        tuples_subj_obj = [[tuple_x[0], tuple_x[2]]
+                           for tuple_x in tuples if len(tuple_x) > 1]
+        tuples_subj_obj_flat = [tuple_x for tuple_pair in tuples_subj_obj
+                                for tuple_x in tuple_pair]
+
+        unique_entities = list(set(tuples_subj_obj_flat))
+        # unique_entities = Counter(tuples_subj_obj_flat).most_common()
```
[the remark this answers is not in the document]
Comment (gallupBenRyan) [reply src/rdf_graph/rdf_graph.py]: Roger - I'm trying here to open the pipe more and get as much of the source data information into the graph as possible.

File: src/rdf_graph/build_graph.py (hunk 2 above)
Comment (gallupBenRyan) [reply src/rdf_graph/build_graph.py]: Good catch - making that import absolute and not relative should be more robust.

**Pull request merged**